tests/app/test_surveys.py

```diff
@@ -68,14 +71,30 @@ def test_surveys_todo(self, mock_request):
         self.assertTrue('RUNAME1_COMPANY4 RUNNAME2_COMPANY4'.encode() in response.data)
 
     @requests_mock.mock()
-    def test_surveys_history(self, mock_request):
-        mock_request.get(url_get_surveys_list, json=surveys_list)
+    def test_surveys_history_seft(self, mock_request):
+        mock_request.get(url_get_surveys_list, json=surveys_list_seft)
+
+        response = self.app.get('/surveys/history')
+
+        self.assertEqual(response.status_code, 200)
+        self.assertTrue('Business Register and Employment Survey'.encode() in response.data)
+        self.assertTrue('RUNAME1_COMPANY4 RUNNAME2_COMPANY4'.encode() in response.data)
+        # Two entries in array, both SEFT. Although 1 is status Complete 2 buttons should show
+        self.assertTrue('ACCESS_SURVEY_BUTTON_1'.encode() in response.data)
+        self.assertTrue('ACCESS_SURVEY_BUTTON_2'.encode() in response.data)
+
+    @requests_mock.mock()
+    def test_surveys_history_eq(self, mock_request):
+        mock_request.get(url_get_surveys_list, json=surveys_list_eq)
 
         response = self.app.get('/surveys/history')
 
         self.assertEqual(response.status_code, 200)
         self.assertTrue('Business Register and Employment Survey'.encode() in response.data)
         self.assertTrue('RUNAME1_COMPANY4 RUNNAME2_COMPANY4'.encode() in response.data)
+        # Two entries in array, one is Complete one is not, so there should be 1 button only
+        self.assertTrue('ACCESS_SURVEY_BUTTON_1'.encode() in response.data)
+        self.assertFalse('ACCESS_SURVEY_BUTTON_2'.encode() in response.data)
```

Comment:
This through me a couple of times reading it, missed the assertFalse. Also could have use assertIn and assertNotIn for better test failure output

Reply:
Made the change as requested - never done Python unit tests before so I just copied what's there.  Someone should probably change all those other ones too ...